lib/puppet_x/certs/common.rb

```diff
@@ -86,8 +86,15 @@ module Common
         newparam(:key_pair) do
           validate do |value|
             param_resource = resource.catalog.resource(value.to_s)
-            if param_resource && !['Puppet::Type::Ca', 'Puppet::Type::Cert'].include?(param_resource.class.to_s)
-              raise ArgumentError, "Expected Ca or Cert resource, got #{param_resource.class} #{param_resource.inspect}"
+
+            if param_resource.is_a?(Puppet::Resource)
+              param_resource_type = param_resource.resource_type
+            else
+              param_resource_type = param_resource.to_resource.resource_type
```

Comment:
This will fail if no `param_resource` was passed in. I think this whole block should be inside `if param_resource`